app/components/dashboard-calendar.js

```diff
@@ -57,232 +62,139 @@ export default Component.extend({
     return 1;
   }),
   calendarDate: momentFormat('selectedDate', 'YYYY-MM-DD'),
-  ourEvents: computed('mySchedule', 'fromTimeStamp', 'toTimeStamp', 'selectedSchool', 'selectedView', function(){
-    if(this.get('mySchedule')) {
-      return DS.PromiseArray.create({
-        promise: this.get('userEvents').getEvents(this.get('fromTimeStamp'), this.get('toTimeStamp'))
-      });
-    } else {
-      let deferred = RSVP.defer();
-      this.get('selectedSchool').then(school => {
-        deferred.resolve(this.get('schoolEvents').getEvents(school.get('id'), this.get('fromTimeStamp'), this.get('toTimeStamp')));
-      });
-      return DS.PromiseArray.create({
-        promise: deferred.promise
+
+  /**
+   * @property academicYears
+   * @type {Ember.computed}
+   * @public
+   */
+  academicYears: computed('allAcademicYears.[]', 'academicYearSelectedByUser', function(){
+    return new Promise(resolve => {
+      this.get('allAcademicYears').then(years => {
+        resolve(years.sortBy('title'));
       });
+    });
+  }),
+
+  /**
+   * @property cohorts
+   * @type {Ember.computed}
+   * @public
+   * @todo Check if this can be replaced with allCohorts(). [ST 2017/02/07]
+   */
+  cohorts: computed('allCohorts.[].displayTitle', 'selectedCohorts.[]', function(){
+    return this.get('allCohorts');
+  }),
+
+  /**
+   * @property courseLevels
+   * @type {Ember.computed}
+   * @public
+   * @todo Review whether this needs to be a computed property, get rid of it if applicable. [ST 2017/02/07]
+   */
+  courseLevels: computed('selectedCourseLevels.[]', function(){
```

Comment:
I agree with you todo.  Seems like this could be done as either a static array or could be built in `init()` if the loop is variable.  CP which depends on something else is weird since this will always return [1,2,3,4,5] no matter what.